spec/index.bs

```diff
@@ -3909,12 +3906,18 @@ A {{GPUBindGroup}} object has the following internal slots:
                         1. Let |internalUsage| be the [=binding usage=] for |layoutBinding|.
                         1. Each [=subresource=] seen by |resource| is added to {{GPUBindGroup/[[usedResources]]}} as |internalUsage|.
                 </div>
-            1. Return |bindGroup|.
 
-            Issue: define the "effective buffer binding size" separately.
+            1. Return |bindGroup|.
         </div>
 </dl>
 
+<div algorithm>
+    <dfn abstract-op>effective buffer binding size</dfn>(binding)
+        1. If |binding|.{{GPUBufferBinding/size}} is `undefined`:
+            1. Return max(0, |binding|.{{GPUBufferBinding/buffer}}.{{GPUBuffer/[[size]]}} - |binding|.{{GPUBufferBinding/offset}});
```

Comment:
I kind of would like if we didn't have to actually write the max() since I don't think that should ever be valid (there's another rule that would disallow it). But since the validation rules are unordered is it weird to assume ("assert") that here?

```suggestion
            1. Assert |binding|.{{GPUBufferBinding/offset}} &lt; |binding|.{{GPUBufferBinding/buffer}}.{{GPUBuffer/[[size]]}}.
            1. Return |binding|.{{GPUBufferBinding/buffer}}.{{GPUBuffer/[[size]]}} - |binding|.{{GPUBufferBinding/offset}};
```

Reply:
We don't actually have a rule that says that we must have `offset < buffer.size`. Only the following:

> The bound part designated by resource.offset and resource.size resides inside the buffer.

So we kind of need to compute the size before using 'effective size' there.

That's why I think the style with "all the following must be true" can't work everywhere because it makes it hard to perform computations in the middle of validation.

Reply:
Yeah, that's the rule I meant. It's too handwavy right now, but it covers it in spirit.

I guess we'll figure out a better way to write this later.

Reply:
```suggestion
            1. Return max(0, |binding|.{{GPUBufferBinding/buffer}}.{{GPUBuffer/[[size]]}} - |binding|.{{GPUBufferBinding/offset}});

                Issue: The `max` is only needed here because nothing strictly guards `offset` from being too large.
                Consider refactoring so it's guarded.
```

Reply:
Talked with @toji and decided not to bother adding the suggested issue. Max is fine.